src/main.rs

```diff
@@ -106,12 +110,32 @@ fn main() -> Result<(), Box<dyn Error>> {
             .arg("log")
             .arg("--numstat")
             .output()?;
+
         String::from_utf8(process.stdout)?
     } else if matches.opt_present("input") {
         let path = matches.opt_str("input").unwrap();
+
         std::fs::read_to_string(path)?
+    } else if matches.opt_present("stdin") {
+        let mut input = String::new();
+        loop {
+            let mut buffer = String::new();
+            match std::io::stdin().read_line(&mut buffer) {
```

Comment:
And you don't like, that it breaks?

Reply:
I just do not know how to handle the edge case which triggers the unintended behaviour.

In summary, there are three cases to take about:

1. The application is listening and there is something to read.
2. The application is not listening and there is something to read.
3. The application is listening and there is nothing to read.

Case one is implemented.  In case two, the application just ignores the input; this is not that nice but at least no bug arising from the implementation.

The question is:  how to handle the third case?

Reply:
To clarify the meaning of "break":  the application does not stop, it continues to wait for input.  I would summarise this unintended behaviour as "broken" which causes the second sentence of the first post due to the used tense of the Indicative.  So, "break" in the meaning of "shattered".

Yeah, it is not the best summary in the context of a `break` instruction but common when discussing a bug.

Reply:
I think `read_line` is blocking, so this case actually only happens if a `EOF` is encountered. And in that case it seems perfectly fine to break out of the loop?

I mean, sure, it looks confusing to the user, if they run the program with `--stdin` and nothing happens, but they should probably look up, what that option does in the first place.

Reply:
We could also leave this as an open point for a further enhancement; for instance, one might use it as a shell to process the commit analysis with more control regarding the possible settings.